pkg/api/norman/store/app/store.go

```diff
@@ -175,3 +187,20 @@ func (s *Store) parseAppExternalID(data map[string]interface{}) (string, string,
 	}
 	return templateVersionID, ns, nil
 }
+
+func (s *Store) parseClusterName(url string) (string, error) {
+	url = strings.TrimPrefix(url, "/v3/project/")
+	url = strings.TrimPrefix(url, "/v3/projects/")
+	parts := strings.Split(url, ":")
+	if !(len(parts) > 1) {
+		return "", httperror.NewAPIError(httperror.InvalidBodyContent, "cannot determine target cluster")
+	}
+	if _, err := s.ClusterLister.Get("", parts[0]); err != nil {
+		if errors.IsNotFound(err) {
+			return "", httperror.NewAPIError(httperror.NotFound, fmt.Sprintf("cannot find target cluster [%s]: %v", parts[0], err))
+		}
+		return "", httperror.NewAPIError(httperror.ServerError, fmt.Sprintf("error getting target custer [%s]: %v", parts[0], err))
+
+	}
+	return parts[0], nil
+}
```

Comment:
Can we not get project name at all from apiContext and then use `ref.parse` ? There is also a `GetClusterID` method that might work. Adding all these things feels dangerous.

Reply:
it's because this is a project scoped api

Reply:
@rmweir I don't understand your answer, why can't `GetClusterID` be used here?

Reply:
@dramich that function is intended to be used on cluster requests. It does not work because the url in this case is of the form: `/v3/project(s)/<clusterName>:<projectName>/app`, but it is expecting `/v3|k8s/cluster(s)/<clusterName>`. I could modify GetClusterID so it will work.